### ISSUE TITLE
Create a reusable "Input" component

### ISSUE THREAD

**Issue by sagarvxyz** [oslabs-beta/Sveste#110]
Right now, user input fields are styled in a Block component, and markup is written in child components.

Instead, We should have a distinct, reusable Input component that:
1. Accepts a parameter for label text
2. Uses that label text as an `id`, `name`, and `for` attribute for `<input>` and `<label>` HTML tags
3. Binds the `value` of the input to the appropriate object in our application store
4. Has markup for a label + input following the current pattern
5. Includes the current styling, specifically a floating label that shrinks to the top left of the input box when a user starts typing in the box

Basically, the Input component should be a reusable component that can be used to create form elements.
